src/msg/async/ProtocolV1.cc

```diff
@@ -15,8 +15,9 @@
 #define dout_prefix _conn_prefix(_dout)
 ostream &ProtocolV1::_conn_prefix(std::ostream *_dout) {
   return *_dout << "--1- " << messenger->get_myaddrs().legacy_addr() << " >> "
-                << connection->peer_addrs->legacy_addr() << " conn("
-                << connection << (connection->msgr2 ? " msgr2" : " legacy")
+                << connection->peer_addrs->legacy_addr() << " " << this
+		<< " conn("
+                << connection << " " << connection->protocol.get()
```

Comment:
we can remove the `connection->protocol.get()` as we are already printing `this` two lines above. Same for V2.

Reply:
i did this deliberately because i was worried there was a bug in the replace() where the protocol relationship was mixed up.. but i may have misread the code, maybe protocol is never moved between Connections?

Reply:
yes, the protocol is never moved between connections.